lib/rubygems/spec_fetcher.rb

```diff
@@ -184,16 +184,24 @@ def suggest_gems_from_name(gem_name, type = :latest, num_results = 5)
 
     matches = names.filter_map do |n|
       len = n.name.length
-      # If the length is min_length or shorter, we've done `max` deletions.
-      # If the length is max_length or longer, we've done `max` insertions.
-      # These would both be rejected later, so we skip early for performance.
-      next if len <= min_length || len >= max_length
-
       # If the gem doesn't support the current platform, bail early.
       next unless n.match_platform?
 
-      # The candidate name, normalized the same as gem_name.
+      # If the length is min_length or shorter, we've done `max` deletions.
+      # This would be rejected later, so we skip it for performance.
+      next if len <= min_length
+
+      # The candidate name, all lowercase like gem_name.
       normalized_name = n.name.downcase
+
+      next [n.name, 0] if ["#{gem_name}-ruby", "ruby-#{gem_name}"].include?(normalized_name)
```

Comment:
Since we do it so many times, just duplicate this line to avoid creating 1000s of arrays.

```suggestion
      next [n.name, 0] if normalized_name == "#{gem_name}-ruby"
      next [n.name, 0] if normalized_name == "ruby-#{gem_name}"
```